15728912 only show published videos

alvaromart opened the pull request:
Added new editor action, publish video. 
Show only published videos if user is not editor or if he is not querying his own videos.
Icon published video showed on top-right of video card with circular background and in video details it is activated.
Upload video redirect to user gallery, main gallery only show published videos.
In video details, user card show video uploaded, not video published.

alvaromart committed:
Added checkbox in edit page to publish videos
- pages/video-detail-edit/partials/aditional-details.html, added md-checkbox publish
- languages EN, ES added VIDEO_EDIT_PUBLISH

alvaromart committed:
Added new editor action, publish video. Show only published videos if user is not editor or if he is not querying his own videos
- components/video-card/video-card.js Added published field to scope and function showPublishedIcon to manage logic.
- components/video-card/video-card.scss Added video-published, added circular background, color white. Update video-time.
- components/video-card/video-card.scss Added span video-published, md-icon cloud_done showed if showPublishedIcon
- language EN, ES added VIDEO_PUBLISHED_CAPTION,
- pages/video-detail/video-detail.controller.js added showPublishedIcon. Only show icon and text if video is published.
- pages/video-detail/video-detail.view.html  Added published-icon, md-icon cloud_done and text VIDEO_PUBLISHED_CAPTION
- pages/video-detail/video-detail.view.scss Added published-icon, color grey, same that pin_person icon.

alvaromart committed:
Upload video redirect to user gallery, gallery only show published videos.
-pages/upload/upload.controller.js Redirect to user gallery, needed param userId.

DevStarlight commented on pages/upload/upload.controller.js:
```diff
@@ -42,7 +42,8 @@ function UploadController(api, session, $state, $translate, productType) {
 			api.upload(api.url + '/video', self.file, data, function() {
 				self.loading = false;
-				$state.go('gallery');
+				console.log('video updated, redirecting to userGallery');
```
I'd say we don't need that console.log anymore =)

alvaromart replied:
Thanks.

DevStarlight commented on components/video-card/video-card.js:
```diff
@@ -16,6 +17,16 @@ function videoCardDirective() {
+		if (session.role === 'editor' || (owner === session.id && $state.current.name === 'userGallery')) {
+			return (published === true || published === 'true');
+		} else {
+			return false;
```
we can improve this code that way: 
```
if (session.role === 'editor' || (owner === session.id && $state.current.name === 'userGallery')) {
   return (published === true || published === 'true');
}
return false;
```

alvaromart replied:
+1, in video-detail-controller we have the same mistake.

alvaromart committed:
Fix PR clean code comments. Update user card video count string to show video uploaded, not video published
- components/video-card/video-card.js Change PUBLISHED_VIDEOS to UPLOADED_VIDEOS
- components/video-card/video-card.js Simplify if/else code.
- languages ES, EN added UPLOADED_VIDEOS
- pages/upload/upload.controller.js Remove console.log
- pages/video-detail/video-detail.controller.js Simplify if/else code.

DevStarlight committed:
Merge branch 'dev' of github.com:Videona/platform-frontend into 15728912_only_show_published_videos

# Conflicts:
#	pages/video-detail/video-detail.controller.js
#	pages/video-detail/video-detail.view.html
#	pages/video-detail/video-detail.view.scss

alvaromart committed:
Merge branch '15728912_only_show_published_videos' of github.com:Videona/platform-frontend into 15728912_only_show_published_videos

DevStarlight committed:
setup v0.11.0 - As Junior I want the gallery to only show published videos
